pkg/client/unversioned/request.go

```diff
@@ -642,6 +642,7 @@ func (r *Request) Watch() (watch.Interface, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
+		defer resp.Body.Close()
```

Comment:
I'd rather keep it here to make it easy to confirm that the body is closed in the same place we create it